### PR TITLE
Fix for GHC 9.2.1 and GHC 9.0.1

### PR DESCRIPTION
**Pull request by matsubara0507**
Occur various errors when build project using GHC 9.2.1 or GHC 9.0.1.

So, fix rules_haskell.

FYI: check by sample project https://github.com/matsubara0507/example-haskell-with-bazel/actions/runs/1593848398

### 1. invalid `stripPrefix`

Use GHC 9.0.1 in Windows: 

```
ERROR: An error occurred during the fetch of repository 'rules_haskell_ghc_windows_amd64':
   Traceback (most recent call last):
	File "C:/users/runneradmin/_bazel_runneradmin/ugsis23d/external/rules_haskell/haskell/ghc_bindist.bzl", line 391, column 29, in _ghc_bindist_impl
		ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error extracting C:/users/runneradmin/_bazel_runneradmin/ugsis23d/external/rules_haskell_ghc_windows_amd64/temp656765299240579188/ghc-9.0.1-x86_64-unknown-mingw32.tar.xz to C:/users/runneradmin/_bazel_runneradmin/ugsis23d/external/rules_haskell_ghc_windows_amd64/temp656765299240579188: Prefix "ghc-9.0.1" was given, but not found in the archive. Here are possible prefixes for this archive: "ghc-9.0.1-x86_64-unknown-mingw32".
```

`ghc-9.0.1-x86_64-unknown-mingw32.tar.xz` is extracted to `ghc-9.0.1-x86_64-unknown-mingw32` not to `ghc-9.0.1` directory. So, fix `stripPrefix` 4531a7f6442dca44b9c4edfdad1ee3c1ce53568f

And, GHC 9.2.1 in macOS and GHC 9.2.1 in Windows is occur same error too 341a03a32fd920d13338e9041c9c946c57cf71db

### 2. Change path `html/libraries/base-4.14.0.0` to `html/libraries/base` with GHC 9.0.1 and GHC 9.2.1

```
ERROR: /path/to/external/rules_haskell_ghc_darwin_amd64/BUILD:23:18: in _haskell_toolchain rule @rules_haskell_ghc_darwin_amd64//:toolchain-impl: 
Traceback (most recent call last):
	File "/path/to/external/rules_haskell/haskell/toolchain.bzl", line 180, column 17, in _haskell_toolchain_impl
		fail("Could not infer `docdir_path` from provided `docdir` attribute. Missing `lib/settings` file.", "docdir")
Error in fail: Could not infer `docdir_path` from provided `docdir` attribute. Missing `lib/settings` file. docdir
```

fix it fde80bd6d1c7126b6df331bab470a9972088b92f

### 3. Change path `lib/package.conf.d` to `lib/lib/package.conf.d` with GHC 9.2.1 in macOS

fix it dc9e0b18278bf6d46c6b7e67f7579d044af655d7

### 4. Change `$topdir` to `${pkgroot}` in pkgdb file with GHC 9.0.1 and GHC 9.2.1

fix it f4c84c423b92f8eb849ad5677d0f1b86e7998055

### 5. `realpath` is not found with GHC 9.2.1 in macOS

See https://gitlab.haskell.org/ghc/ghc/-/issues/19963
So, patch [this commit](https://gitlab.haskell.org/ghc/ghc/-/commit/fab2579e63bb317d4c266d7b949cf96ad6e5d17b).

### 6. `--enable-relocatable` is not work from Cabal 3.6.0

So, disable `--enable-relocatable` when GHC 9.2.1 (Cabal 3.6.0) 20085b2af090645d1daae15dd910964dad255611
I think ideally disable `--enable-relocatable` only when use Cabal 3.6.0. But I don't have idea method get Cabal package version.

### 7.  Invalid `haddock-html` path in pkgdb with GHC 9.2.1 and GHC 9.0.1 in Windows

fix it using patch e22e9d286247ce6b2f6fbc199448c57f57b4c5ff 1b70303402fe372464a9c69cce70ac30ac8d4253

### 8. Change `doc` to `docs` with GHC 9.2.1 and GHC 9.0.1 in Windows

fix it 4ce551ca5e99b5b579cc317c401498f08b48f299 cf3082ad863597ae19075bb779881eae4202cc8e

### CODE REVIEW

**Review by aherrmann — COMMENTED** [review]
Thank you for contributing these!
I'll defer to @Xophmeister for review.

**Comment by Xophmeister** [comment]
This all looks good to me :+1: There was a transient problem with the CI; it's rerunning now. Once it's all green, I'll merge.

Thanks for the contribution :slightly_smiling_face:

**Review by Xophmeister — APPROVED** [review]

**Pull request merged**
